identifiable/src/main/java/com/r3/corda/ledger/utxo/identifiable/IdentifiableConstraints.java

```diff
@@ -27,10 +27,13 @@ public final class IdentifiableConstraints {
             "On identifiable state(s) updating, at least one identifiable state must be created.";
 
     final static String CONTRACT_RULE_UPDATE_IDENTIFIERS =
-            "On identifiable state(s) updating, only one identifiable state with a matching identifier must be consumed for every created identifiable state with a non-null identifier.";
+            "On identifiable state(s) updating, a produced identifiable state must exist exactly once as a consumed identifiable state when the identifier is not null.";
 
-    final static String CONTRACT_RULE_UPDATE_IDENTIFIER_EXCLUSIVITY =
-            "On identifiable state(s) updating, every created identifiable state's identifier must appear only once when the identifier is not null.";
+    final static String CONTRACT_RULE_UPDATE_INPUT_IDENTIFIER_EXCLUSIVITY =
+            "On identifiable state(s) updating, a consumed identifiable state must exist exactly once as an input when the identifier is not null.";
+
+    final static String CONTRACT_RULE_UPDATE_OUTPUT_IDENTIFIER_EXCLUSIVITY =
+            "On identifiable state(s) updating, a created identifiable state must exist exactly once as an output when the identifier is not null.";
```

Comment:
Then to align with the refactor above:

"On identifiable state(s) updating, duplicate identifiable state identifiers must not be created."